addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXBridgeBaseThingHandler.java

```diff
@@ -0,0 +1,974 @@
+import org.openhab.binding.knx.GroupAddressListener;
+import org.openhab.binding.knx.IndividualAddressListener;
+import org.openhab.binding.knx.KNXBindingConstants;
+import org.openhab.binding.knx.KNXBusListener;
+import org.openhab.binding.knx.internal.dpt.KNXCoreTypeMapper;
+import org.openhab.binding.knx.internal.dpt.KNXTypeMapper;
+import org.openhab.binding.knx.internal.logging.LogAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tuwien.auto.calimero.CloseEvent;
+import tuwien.auto.calimero.DetachEvent;
+import tuwien.auto.calimero.FrameEvent;
+import tuwien.auto.calimero.GroupAddress;
+import tuwien.auto.calimero.IndividualAddress;
+import tuwien.auto.calimero.cemi.CEMILData;
+import tuwien.auto.calimero.datapoint.CommandDP;
+import tuwien.auto.calimero.datapoint.Datapoint;
+import tuwien.auto.calimero.exception.KNXException;
+import tuwien.auto.calimero.exception.KNXIllegalArgumentException;
+import tuwien.auto.calimero.exception.KNXRemoteException;
+import tuwien.auto.calimero.exception.KNXTimeoutException;
+import tuwien.auto.calimero.link.KNXLinkClosedException;
+import tuwien.auto.calimero.link.KNXNetworkLink;
+import tuwien.auto.calimero.link.NetworkLinkListener;
+import tuwien.auto.calimero.log.LogManager;
+import tuwien.auto.calimero.mgmt.Destination;
+import tuwien.auto.calimero.mgmt.KNXDisconnectException;
+import tuwien.auto.calimero.mgmt.ManagementClient;
+import tuwien.auto.calimero.mgmt.ManagementClientImpl;
+import tuwien.auto.calimero.mgmt.ManagementProcedures;
+import tuwien.auto.calimero.mgmt.ManagementProceduresImpl;
+import tuwien.auto.calimero.process.ProcessCommunicator;
+import tuwien.auto.calimero.process.ProcessCommunicatorImpl;
+import tuwien.auto.calimero.process.ProcessEvent;
+import tuwien.auto.calimero.process.ProcessListenerEx;
+
+/**
+ * The {@link KNXBridgeBaseThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Kai Kreuzer / Karel Goderis - Initial contribution
+ */
+public abstract class KNXBridgeBaseThingHandler extends BaseThingHandler {
+
+    public final static int ERROR_INTERVAL_MINUTES = 5;
+
+    protected Logger logger = LoggerFactory.getLogger(KNXBridgeBaseThingHandler.class);
+
+    private Collection<GroupAddressListener> groupAddressListeners = new HashSet<GroupAddressListener>();
+    private Collection<IndividualAddressListener> individualAddressListeners = new HashSet<IndividualAddressListener>();
+    private Collection<KNXBusListener> knxBusListeners = new HashSet<KNXBusListener>();
+    static protected Collection<KNXTypeMapper> typeMappers = new HashSet<KNXTypeMapper>();
+    private LinkedBlockingQueue<RetryDatapoint> readDatapoints = new LinkedBlockingQueue<RetryDatapoint>();
+    protected ConcurrentHashMap<IndividualAddress, Destination> destinations = new ConcurrentHashMap<IndividualAddress, Destination>();
+
+    protected ItemChannelLinkRegistry itemChannelLinkRegistry;
```

Comment:
In fact, for the KNX binding you might. This was discussed several times in the past, whereby one has to decide what the binding will do wrt "bridging" values between Items.